### ISSUE TITLE
Is it worth it to test the model on Spider dataset's images

### ISSUE THREAD

**Issue by abelsalm** [ivadomed/model-canal-seg#18]
As we discussed in last dcm normalization meeting (2024/10/30), when asking "where to find some data with accurate groundtruth segmentations to test the model", one answer was to try it on Spider. The other idea was to create new groundtruths with sct_propseg on datasets we're interested in for a validation. 

So the main advantage is that the spider segmentations are already available, on 257 subjects, but i's only lumbar, and not perfectly segmented : all of the images are over segmented, for differents reasons, see below

Here for instance (canal should be limited by the green line)
<img width="698" alt="image" src="https://github.com/user-attachments/assets/3205f300-8733-4c44-8db9-95aa3a98c3e6">

Or here, white voxels are segmented besides of being out of the dural sac, which is the limit we defined for the spinal canal
<img width="959" alt="image" src="https://github.com/user-attachments/assets/e6b96c27-7258-4ec1-b1f5-90dffe74c997">

On the other side, I could generate some segmentations with sct_propseg, but it's quite time consuming for the correction part. However it would enable to create a validation set mixing differents datasets and with more accurate groundtruth segmentations. 

So my question here would be : what should I choose ? Knowing that both is also possible, and considering that I would use those validaiton methods for the article about the model.

**Comment by abelsalm** [comment]
@jcohenadad @sandrinebedard @valosekj @NathanMolinier if you have any suggestions !

**Comment by jcohenadad** [comment]
beyond the 'proper' anatomy, we should also be realistic about how the DL algorithm works, and what researchers will do with the spinal canal segmentation.

to me, that segmentation below looks ok:

![image](https://github.com/user-attachments/assets/d9e409c9-ced9-4df3-9d03-a200bc37401a)

**Comment by valosekj** [comment]
> I could generate some segmentations with sct_propseg, but it's quite time consuming for the correction part. However it would enable to create a validation set mixing differents datasets and with more accurate groundtruth segmentations.

I tried `sct_propseg -i sub-amuAP_T2w.nii.gz -c t2 -CSF` on a random subject (`sub-amuAP`) from the `whole-spine` dataset, and the canal segmentation is not too bad (of course, some corrections are needed, but hopefully it should not be longer than 10 minutes/subject). So a possible approach could be to select 5-6 images that were **_not_** included in the training/validation sets (e.g., 2 from whole-spine, 2 from spine-generic, 2 from [ds004507](https://openneuro.org/datasets/ds004507/versions/1.0.1)), run `sct_propseg`, and correct.

<img width="1060" alt="image" src="https://github.com/user-attachments/assets/892022ed-2cd2-4a3c-8988-788097d0ffb8">

**Comment by sandrinebedard** [comment]
these ar eteh GT that @NathanMolinier used for total spine seg no?

**Comment by NathanMolinier** [comment]
Yes indeed, but they are not perfect I think since they were generated by registering the PAM50 template to each subject space.

Another idea would be to compare your model with totalspineseg to show that your model is more accurate.
